### PR TITLE
Sam/com 219 rename lists

### PR DESCRIPTION
**Pull request by asifmu97**
This branch should reflect the following:

1. Rename list functionality on MyLists, User DSO, services
2. The dialog box on ListNav (when at a certain list) has its spacing adjusted
3. The dialog box on AllIngredients has also been adjusted in terms of vertical spacing
4. Renaming and editing lists is now case-sensitive. That is, you can have "Grocery" and "grocery" in your list. This is done to 
mimic the backend.

**NOTE: please download the latest env file for the endpoint. The variables have also been spaced out and grouped as per @h-ng 's suggestion**  😄

### CODE REVIEW

File: src/services/Backend.ts [hunk 1]
```diff
@@ -318,4 +318,28 @@ export class Backend implements BackendInterface {
...
+					headers: { Authorization: "Bearer " + token },
+				}
+			);
+
+            if (response.status === 200) {
+                this.userAuth.setListName(oldName, newName);
+            } else {
+                console.error(`Error: Received status code ${response.status}`);
+            }
+        }catch (error) {
```
Comment (gapy04) [inline src/services/Backend.ts:341]: add space after } so that it's consistent with other code

File: src/models/List.ts [hunk 2]
```diff
@@ -32,4 +32,7 @@ export class List {
         this.ingredients.splice(ingredientIndex, 1);
     }
 
+    setListName(newName:string):void{
```
Comment (gapy04) [inline src/models/List.ts:35]: Add spaces to be consistent with other parts of the code.
`(newName: string): void {`

File: src/components/MyLists/MyLists.tsx [hunk 3]
```diff
@@ -335,6 +411,51 @@ function MyLists({ userAuth, backendInterface }: MyListsProps) {
...
+                    error={!!nameError}
+                    helperText={nameError}
+                />
+                </DialogContent>
+                <DialogActions>
+                    <Button
+                        onClick={() => handleRenameList(oldName, newName)}
+                        disabled={!newName.trim()}
+                        className="secondary-color"
+                        style={{color: "white"}}
```
Comment (gapy04) [inline src/components/MyLists/MyLists.tsx:447]: Just add spacing here to match other parts of the code. {{ color: "white" }}

**Review by gapy04 — CHANGES_REQUESTED** [review]
Just formatting changes (adding spaces) to make sure code format matches.

File: src/services/Backend.ts (hunk 1 above)
Comment (asifmu97) [reply src/services/Backend.ts:341]: done!

**Review by gapy04 — APPROVED** [review]
Nicely done!

**Pull request merged**